ubuntu22.04/precompiled/nvidia-driver

```diff
@@ -210,10 +211,26 @@ _unload_driver() {
 
 # Link and install the kernel modules from a precompiled packages
 _install_driver() {
-    echo "Installing NVIDIA driver kernel modules..."
-    apt-get install --no-install-recommends -y linux-objects-nvidia-${DRIVER_BRANCH}-server-${KERNEL_VERSION} \
-    linux-signatures-nvidia-${KERNEL_VERSION} \
-	linux-modules-nvidia-${DRIVER_BRANCH}-server-${KERNEL_VERSION}
+    if [ "$OPEN_KERNEL_MODULES_ENABLED" = true ]; then
+        echo "Installing Open NVIDIA driver kernel modules..."
+        apt-get install -y  linux-signatures-nvidia-${KERNEL_VERSION}
+        # Installing linux-signatures-nvidia-${KERNEL_VERSION} and linux-objects-nvidia-${DRIVER_BRANCH}-server-open-${KERNEL_VERSION}
+        # This auto-installs linux-modules-${KERNEL_VERSION}-generic and conflicts with linux-modules-nvidia-${DRIVER_BRANCH}-server-open-${KERNEL_VERSION}
+        # Using dpkg -i --force-overwrite to resolve the conflict
+        # https://forums.developer.nvidia.com/t/ubuntu-and-nvidia-provided-packages-conflict-breaking-installation/259150
+        apt-get install --download-only --no-install-recommends -y linux-objects-nvidia-${DRIVER_BRANCH}-server-open-${KERNEL_VERSION} && \
+            dpkg -i --force-overwrite $(find /var/cache/apt/archives/ -name "linux-objects-nvidia-${DRIVER_BRANCH}-server-open-${KERNEL_VERSION}_*.deb")
+        apt-get install --download-only --no-install-recommends -y linux-modules-nvidia-${DRIVER_BRANCH}-server-open-${KERNEL_VERSION} && \
+            dpkg -i --force-overwrite $(find /var/cache/apt/archives/ -name "linux-modules-nvidia-${DRIVER_BRANCH}-server-open-${KERNEL_VERSION}_*.deb")
+        # Ensure modules are correctly loaded
+        depmod -a ${KERNEL_VERSION}
```

Comment:
```suggestion
        apt-get install --no-install-recommends -y \
            linux-signatures-nvidia-${KERNEL_VERSION} \
            linux-modules-nvidia-${DRIVER_BRANCH}-server-open-${KERNEL_VERSION}
```

Reply:
linux-signatures-nvidia-${KERNEL_VERSION}   also have conflict with open  linux-modules-nvidia* driver

```
dpkg-deb -c linux-signatures-nvidia-5.15.0-112-generic_5.15.0-112.122+1_amd64.deb
drwxr-xr-x root/root         0 2024-06-14 02:41 ./
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-390/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-390/bits/
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-390/bits/nvidia-drm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-390/bits/nvidia-modeset.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-390/bits/nvidia-uvm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-390/bits/nvidia.ko.sig
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-450srv/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-450srv/bits/
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-450srv/bits/nvidia-drm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-450srv/bits/nvidia-modeset.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-450srv/bits/nvidia-uvm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-450srv/bits/nvidia.ko.sig
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470/bits/
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470/bits/nvidia-drm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470/bits/nvidia-modeset.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470/bits/nvidia-peermem.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470/bits/nvidia-uvm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470/bits/nvidia.ko.sig
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470srv/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470srv/bits/
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470srv/bits/nvidia-drm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470srv/bits/nvidia-modeset.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470srv/bits/nvidia-peermem.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470srv/bits/nvidia-uvm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-470srv/bits/nvidia.ko.sig
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535/bits/
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535/bits/nvidia-drm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535/bits/nvidia-modeset.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535/bits/nvidia-peermem.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535/bits/nvidia-uvm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535/bits/nvidia.ko.sig
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535srv/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535srv/bits/
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535srv/bits/nvidia-drm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535srv/bits/nvidia-modeset.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535srv/bits/nvidia-peermem.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535srv/bits/nvidia-uvm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-535srv/bits/nvidia.ko.sig
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550/bits/
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550/bits/nvidia-drm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550/bits/nvidia-modeset.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550/bits/nvidia-peermem.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550/bits/nvidia-uvm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550/bits/nvidia.ko.sig
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550srv/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550srv/bits/
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550srv/bits/nvidia-drm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550srv/bits/nvidia-modeset.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550srv/bits/nvidia-peermem.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550srv/bits/nvidia-uvm.ko.sig
-rw-r--r-- root/root       790 2024-06-14 02:41 ./lib/modules/5.15.0-112-generic/kernel/nvidia-550srv/bits/nvidia.ko.sig
drwxr-xr-x root/root         0 2024-06-14 02:41 ./usr/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./usr/share/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./usr/share/doc/
drwxr-xr-x root/root         0 2024-06-14 02:41 ./usr/share/doc/linux-signatures-nvidia-5.15.0-112-generic/
-rw-r--r-- root/root       443 2024-06-14 02:41 ./usr/share/doc/linux-signatures-nvidia-5.15.0-112-generic/changelog.Debian.gz
-rw-r--r-- root/root     11251 2024-06-14 02:41 ./usr/share/doc/linux-signatures-nvidia-5.15.0-112-generic/copyright
```

Reply:
verified and It is working fine